feature/google/src/testUnityIAP/java/BillingWrapperBC4Test.kt

```diff
@@ -1,123 +0,0 @@
-package com.revenuecat.purchases.google
-
-import androidx.test.ext.junit.runners.AndroidJUnit4
-import com.android.billingclient.api.BillingClient
-import com.android.billingclient.api.PurchaseHistoryResponseListener
-import com.android.billingclient.api.PurchasesResponseListener
-import com.revenuecat.purchases.ProductType
-import io.mockk.every
-import io.mockk.slot
-import org.assertj.core.api.Assertions.assertThat
-import org.assertj.core.api.AssertionsForClassTypes.fail
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.robolectric.annotation.Config
-import java.util.concurrent.CountDownLatch
-
-@RunWith(AndroidJUnit4::class)
-@Config(manifest = Config.NONE)
-class BillingWrapperBC4Test : BillingWrapperTestBase() {
-
-    @Test
-    fun `queryPurchaseHistoryAsync only calls one response when BillingClient responds twice`() {
-        var numCallbacks = 0
-
-        val slot = slot<PurchaseHistoryResponseListener>()
-        every {
-            mockClient.queryPurchaseHistoryAsync(
-                any(),
-                capture(slot)
-            )
-        } answers {
-            slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
-            slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
-        }
-
-        wrapper.queryPurchaseHistoryAsync(
-            BillingClient.SkuType.SUBS,
-            {
-                numCallbacks++
-            }, {
-                fail("shouldn't be an error")
-            })
-
-        assertThat(numCallbacks == 1)
-    }
-
-    @Test
-    fun `queryPurchaseHistoryAsync only calls one response when BillingClient responds twice from different threads`() {
-        var numCallbacks = 0
-
-        val slot = slot<PurchaseHistoryResponseListener>()
-        val lock = CountDownLatch(2)
-        every {
-            mockClient.queryPurchaseHistoryAsync(
-                any(),
-                capture(slot)
-            )
-        } answers {
-            Thread {
-                slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
-                lock.countDown()
-            }.start()
-
-            Thread {
-                slot.captured.onPurchaseHistoryResponse(billingClientOKResult, null)
-                lock.countDown()
-            }.start()
-        }
-
-        wrapper.queryPurchaseHistoryAsync(
-            BillingClient.SkuType.SUBS,
-            {
-                // ensuring we don't hit an edge case where numCallbacks doesn't increment before the final assert
-                handler.post {
-                    numCallbacks++
-                }
-            }, {
-                fail("shouldn't be an error")
-            })
-
-        lock.await()
-        assertThat(lock.count).isEqualTo(0)
-
-        assertThat(numCallbacks).isEqualTo(1)
-    }
-
-    @Test
-    fun `getPurchaseType returns UNKNOWN if sub not found and inapp responses not OK`() {
```

Comment:
We don't have a test for this in the BC5Test, do we still need it? Also that file `BillingWrapperBC5Test.kt` should be `BillingWrapperTest.kt`

Reply:
i consolidated that file in the unit test PR, is it okay to keep that there?

but let me make sure we aren't missing the test...

Reply:
great catch, we are missing it...i didn't have a way to mock different responses for each type. updated that, though i can't really rerun the tests to be sure it works since there are a ton of build errors. just something to keep in mind when we fix unit tests

Reply:
sure, even just having the empty test with a TODO works